android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt

```diff
@@ -209,42 +216,52 @@ class XMTPModule : Module() {
             }
         }
 
+        AsyncFunction("requestMessageHistorySync") Coroutine { inboxId: String ->
+            withContext(Dispatchers.IO) {
+                val client = clients[inboxId] ?: throw XMTPException("No client")
+                client.requestMessageHistorySync()
+            }
+        }
+
         //
         // Auth functions
         //
-        AsyncFunction("auth") { address: String, environment: String, appVersion: String?, hasCreateIdentityCallback: Boolean?, hasEnableIdentityCallback: Boolean?, enableV3: Boolean?, dbEncryptionKey: List<Int>?, dbDirectory: String? ->
-            logV("auth")
-            val reactSigner = ReactNativeSigner(module = this@XMTPModule, address = address)
-            signer = reactSigner
-
-            if (hasCreateIdentityCallback == true)
-                preCreateIdentityCallbackDeferred = CompletableDeferred()
-            if (hasEnableIdentityCallback == true)
-                preEnableIdentityCallbackDeferred = CompletableDeferred()
-            val preCreateIdentityCallback: PreEventCallback? =
-                preCreateIdentityCallback.takeIf { hasCreateIdentityCallback == true }
-            val preEnableIdentityCallback: PreEventCallback? =
-                preEnableIdentityCallback.takeIf { hasEnableIdentityCallback == true }
-            val context = if (enableV3 == true) context else null
-            val encryptionKeyBytes =
-                dbEncryptionKey?.foldIndexed(ByteArray(dbEncryptionKey.size)) { i, a, v ->
-                    a.apply { set(i, v.toByte()) }
-                }
+        AsyncFunction("auth") {
+            { address: String, environment: String, appVersion: String?, hasCreateIdentityCallback: Boolean?, hasEnableIdentityCallback: Boolean?, enableV3: Boolean?, dbEncryptionKey: List<Int>?, dbDirectory: String?, historySyncUrl: String? ->
```

Comment:
8 was apparently the maximum amount of variables that could be passed into the AsyncFunction. The addition of historySyncUrl broke this. So I had to add an extra closure. I need to triple confirm that this doesn't break login for wallets before merging.